packages/core/src/internal-components/overlay/overlay.element.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  AriaRoles,
+  baseStyles,
+  createId,
+  CdsBaseFocusTrap,
+  event,
+  EventEmitter,
+  FocusTrapTracker,
+  internalProperty,
+  onKey,
+  property,
+} from '@clr/core/internal';
+import { html, query } from 'lit-element';
+import { ifDefined } from 'lit-html/directives/if-defined';
+import { styles } from './overlay.element.css.js';
+
+export function isNestedOverlay(myId: string, overlayPrefix: string, trapIds: string[]): boolean {
+  const overlayIds = trapIds.filter(id => id.indexOf(overlayPrefix) > -1);
+  return overlayIds.indexOf(myId) > 0; // id is present and not the first one...
+}
+
+export function overlayIsActive(overlayId: string, focusTrapService = FocusTrapTracker) {
+  return focusTrapService.getCurrentTrapId() === overlayId;
+}
+
+type CloseChangeSources = 'backdrop-click' | 'escape-keypress' | 'close-button-click' | 'custom';
+
+/**
+ *
+ * ```typescript
+ * import '@clr/core/internal-components/overlay/register.js';
+ * ```
+ *
+ * ```html
+ * <cds-internal-overlay>
+ *  <section cds-layout="vertical align:horizontal-stretch">
+ *    <div cds-layout="vertical pad:md gap:md">
+ *      <h2 cds-text="display">A Title</h2>
+ *      <div>
+ *        <p cds-text="body">
+ *          Content inside a generic overlay.
+ *        </p>
+ *      </div>
+ *    </div>
+ *  </section>
+ * </cds-internal-overlay>
+ * ```
+ *
+ * @beta
+ * @element cds-internal-overlay
+ * @slot - Content slot for the content inside the overlay's panel
+ * @event closeChange - Notify user when close event has occurred
+ * @cssprop --backdrop-background
+ * @cssprop --backdrop-opacity
+ * @cssprop --layered-backdrop-opacity
```

Comment:
Once this is rebased with the new themes you may not need these anymore since they are part of the theme under the object tokens.